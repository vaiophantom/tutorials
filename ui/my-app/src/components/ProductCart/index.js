import CONSTANT from "../../constants";
import { Image, Form } from "react-bootstrap";
import { useEffect, useState } from "react";
import { currencyFormat } from "../../utils/helper";

const ProductCart = (props) => {
  const { data, outputCheckBox } = props;
  const { checked } = data;
  const [check, setCheck] = useState(false);

  useEffect(() => {
    setCheck(checked);
  }, [checked]);

  const onChangeCheckBox = (e) => {
    setCheck(e.target.checked);
    outputCheckBox({ ...data, checked: e.target.checked });
  };
  return (
    <div className="product-cart-screen">
      <div
        style={{
          flex: 3,
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
        }}
      >
        <Form.Check
          onChange={onChangeCheckBox}
          checked={check}
          style={{ marginRight: 20 }}
          type="checkbox"
          id={data.id}
        ></Form.Check>
        <Image
          style={{ marginRight: 20 }}
          src={data.img}
          height={100}
          width={100}
        />
        <p>{data.description}</p>
      </div>
      <div style={{ flex: 1 }}>
        <p>{data.price} $ </p>
      </div>
      <div style={{ flex: 1 }}>
        <p>{data.quantity}</p>
      </div>
      <div style={{ flex: 1 }}>
        <p>{currencyFormat(data.price * data.quantity)} $</p>
      </div>
    </div>
  );
};
export default ProductCart;
