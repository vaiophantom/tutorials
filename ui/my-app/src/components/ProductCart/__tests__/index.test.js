import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ProductCart from '../';
import CONSTANT from '../../../constants';

const mockProps = {
    data: {...CONSTANT.PRODUCT_CART, checked: false},
    outputCheckBox: jest.fn()
}

it('renders Product Cart', async () => {
 const {container} = render(<ProductCart {...mockProps}/>, {wrapper: BrowserRouter});
 expect(container).toMatchSnapshot();
});
