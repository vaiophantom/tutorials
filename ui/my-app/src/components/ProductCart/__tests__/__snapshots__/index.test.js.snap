// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Product Cart 1`] = `
<div>
  <div
    class="product-cart-screen"
  >
    <div
      style="flex: 3; display: flex; flex-direction: row; align-items: center;"
    >
      <div
        class=""
        style="margin-right: 20px;"
      >
        <input
          class="form-check-input"
          type="checkbox"
        />
      </div>
      <img
        class=""
        height="100"
        src="https://images.unsplash.com/photo-1505740420928-5e560c06d30e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8M3x8cHJvZHVjdHxlbnwwfHwwfHw%3D&w=1000&q=80"
        style="margin-right: 20px;"
        width="100"
      />
      <p>
        description
      </p>
    </div>
    <div
      style="flex: 1;"
    >
      <p>
        100
         $ 
      </p>
    </div>
    <div
      style="flex: 1;"
    >
      <p>
        1
      </p>
    </div>
    <div
      style="flex: 1;"
    >
      <p>
        100.00
         $
      </p>
    </div>
  </div>
</div>
`;
